### PR TITLE
display contact information in contact and directory according to the…

### PR DESCRIPTION
**Pull request by rabuzarus**
… other contact related pages

Both pages display now if it's an forum account or not
Additionally some detail parts are not shown in the directory (because they would take to much space).

In addition some code beautification took place.
`z_root()` is used instead of `$a->get_baseurl()`

Mike wrote something which about that `z_root()` doesn't deal with self signed ssl certificates (https://github.com/friendica/friendica/pull/2099#issuecomment-160232770)

Can someone step up into this? Otherwise it would be better that I revert the `z_root()` changes.

### CODE REVIEW

**Comment by redmatrix** [comment]
There's a parameter to get_baseurl which if I recall forces SSL on the link, even if the site was accessed with http. This is used primarily on login pages to force logins through ssl. The rest of the time it doesn't matter, and you just use whatever link you used already to access the site, whether it was http or https. z_root just uses the default and shouldn't be used on authentication links if you've got a self_signed cert ssl policy. This can also be easily fixed by adding the $force_ssl variable with default false to z_root and passing it through get_baseurl.

File: mod/contacts.php [hunk 1]
```diff
@@ -855,6 +855,10 @@ function _contact_detail_for_template($rr){
 		$sparkle = '';
 	}
 
+	//test if contact is a forum page
+	if (isset($rr['forum']) OR isset($rr['prv']))
+				$community = ($rr['forum'] OR $rr['prv']);
```
Comment (annando) [inline mod/contacts.php]: Is the variable $community defined before? It should be set to false by default - although the case should not happen that the two array values are undefined.

File: mod/directory.php [hunk 2]
```diff
@@ -93,8 +97,17 @@ function directory_content(&$a) {
 
 		foreach($r as $rr) {
 
+			$community = '';
 
-			$profile_link = $a->get_baseurl() . '/profile/' . ((strlen($rr['nickname'])) ? $rr['nickname'] : $rr['profile_uid']);
+			// get the friendica address and the profile url of the user
+			$p = q("SELECT `addr` AS faddr, `url` AS profile_url FROM `contact` WHERE `uid` = %d AND `self`",
+				intval($rr['uid'])
```
Comment (annando) [inline mod/directory.php]: This query should be done in the query part before the loop. This reduces the amount of queries.

**Comment by rabuzarus** [comment]
@annando 
INNER JOIN or LEFT JOIN?
I would say INNER JOIN because there shouldn't be more contacts as result than contacts with `self`. Am I right?

**Comment by annando** [comment]
Left Join sounds good. But better test it out by yourself in the sql console.

**Comment by rabuzarus** [comment]
Are there any concerns in accepting this? I need the function page_type_translate() which is included in this PR

**Comment by fabrixxm** [comment]
not for me

**Pull request merged**